pkg/chains/signing/kms/kms.go

```diff
@@ -95,12 +98,27 @@ func NewSigner(ctx context.Context, cfg config.KMSSigner) (*Signer, error) {
 	// pass through configuration options to RPCAuth used by KMS in sigstore
 	rpcAuth := options.RPCAuth{
 		Address: cfg.Auth.Address,
-		Token:   cfg.Auth.Token,
+		//Token:   cfg.Auth.Token,
 		OIDC: options.RPCAuthOIDC{
 			Role: cfg.Auth.OIDC.Role,
 			Path: cfg.Auth.OIDC.Path,
 		},
 	}
+
+	// get token from file KMS_AUTH_TOKEN, a mounted secret at signers.kms.auth.token-dir or
+	// as direct value set from signers.kms.auth.token.
+	// If both values are set, priority will be given to token-dir.
+
+	if cfg.Auth.TokenDir != "" {
```

Comment:
The only thing I'm thinking this is should we support for all kms providers or just vault ?

Reply:
There is a possibility for other providers as well to allow token based auth to the keys. The dependent package sigstore signature options as well has it generic.(https://github.com/sigstore/sigstore/blob/main/pkg/signature/options/rpcauth.go#L25) . so I believe it is fine to have it generic.